app/util/rpc-domain-utils.test.ts

```diff
@@ -1,152 +1,377 @@
-      jest.spyOn(rpcUtils, 'extractRpcDomain').mockReturnValue(RpcDomainStatus.Private);
-      expect(generateRPCProperties('0x3')).toEqual({
-        properties: { rpc_domain: RpcDomainStatus.Private },
-        sensitiveProperties: {},
+  });
+  describe('getNetworkRpcUrl', () => {
+    describe('when retrieving RPC URLs', () => {
+      it('returns RPC URL from legacy format', () => {
+        // Setup
+        const { mockNetworkController } = setupTestEnvironment();
+        mockNetworkController.findNetworkClientIdByChainId.mockReturnValue('network1');
+        mockNetworkController.getNetworkConfigurationByNetworkClientId.mockReturnValue({
+          rpcUrl: 'https://legacy-rpc.com',
+        });
+        (Engine.context as unknown as { NetworkController: MockNetworkController }).NetworkController = mockNetworkController;
+        // Exercise
+        const result = getNetworkRpcUrl('0x1');
+        // Verify
+        expect(result).toBe('https://legacy-rpc.com');
+      });
+      it('returns RPC URL from rpcEndpoints array', () => {
+        // Setup
+        const { mockNetworkController } = setupTestEnvironment();
+        mockNetworkController.findNetworkClientIdByChainId.mockReturnValue('network1');
+        mockNetworkController.getNetworkConfigurationByNetworkClientId.mockReturnValue({
+          rpcEndpoints: [
+            { url: 'https://rpc1.com' },
+            { url: 'https://rpc2.com' },
+          ],
+          defaultRpcEndpointIndex: 1,
+        });
+        (Engine.context as unknown as { NetworkController: MockNetworkController }).NetworkController = mockNetworkController;
+        // Exercise
+        const result = getNetworkRpcUrl('0x1');
+        // Verify
+        expect(result).toBe('https://rpc2.com');
+      });
+      it('returns unknown when network client ID not found', () => {
+        // Setup
+        const { mockNetworkController } = setupTestEnvironment();
+        mockNetworkController.findNetworkClientIdByChainId.mockReturnValue(null);
+        (Engine.context as unknown as { NetworkController: MockNetworkController }).NetworkController = mockNetworkController;
+        // Exercise
+        const result = getNetworkRpcUrl('0x1');
+        // Verify
+        expect(result).toBe('unknown');
+      });
+      it('returns unknown when network configuration not found', () => {
+        // Setup
+        const { mockNetworkController } = setupTestEnvironment();
+        mockNetworkController.findNetworkClientIdByChainId.mockReturnValue('network1');
+        mockNetworkController.getNetworkConfigurationByNetworkClientId.mockReturnValue(null);
+        (Engine.context as unknown as { NetworkController: MockNetworkController }).NetworkController = mockNetworkController;
+        // Exercise
+        const result = getNetworkRpcUrl('0x1');
+        // Verify
+        expect(result).toBe('unknown');
+      });
+      it('handles errors gracefully', () => {
```

Comment:
S; this name only tells me it's about errors.
"Handles" can be anything and I have no idea what's your definition of "gracefully".
Try something like:

```suggestion
      it('returns unknown as RPC URL on error', () => {
```